### PR TITLE
DO NOT MERGE - Added three.js math libraries for use in hifi

### PR DESCRIPTION
**Pull request by SeijiEmery**
Made some minor tweaks to make this library work w/ hifi (changed THREE to be a global instead of local var, etc). Can be used as-is w/ Script.include (Three.js has to be included first, then the other files), and includes a better (for js) implementation of 3d and 2d vectors, quaternions, matrices, and a bevy of math utilities for dealing with rays, planes, rects, etc.,

Three's implementation is also 10-30x faster than the hifi versions (benchmarked) since it doesn't have js <--> c++ translation overhead, and this does make a difference for computation-intensive js code (flocking, etc).

Documentation at http://threejs.org/docs/ (this only includes the math library).

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Build console output for commit: 021958f49c45c03214874f28375481dcde6b1d51 available at: https://gist.github.com/15948b6eaeffda210081<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5848.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5848.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: a9fef1a8741e4b5d4f53bc4ba228f6d3d266335e available at: https://gist.github.com/15948b6eaeffda210081<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5848.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5848.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: dabc8ce62a82b03da4b8cbb95f8dc5ee17e88271 available at: https://gist.github.com/15948b6eaeffda210081<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5848.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5848.dmg

File: examples/utilities/tests/threeMathBenchmark.js [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+//  Distributed under the Apache License, Version 2.0.
+//  See the accompanying file LICENSE or http://www.apache.org/licenses/LICENSE-2.0.html
+//
+
+Script.include('perfTest.js');
+Script.include('../../libraries/three/Three.js');
+Script.include('../../libraries/three/math/Vector3.js');
+Script.include('../../libraries/three/math/Quaternion.js');
+
+tryCatch(function () {
```
Comment (ZappoMan) [inline examples/utilities/tests/threeMathBenchmark.js]: this file certainly should match our coding standard which is 4 space indents. not tabs. please repair.

Comment (SeijiEmery) [inline examples/utilities/tests/threeMathBenchmark.js]: yes, it should -- fixed

**Comment by hifi-gustavo** [comment]
Build console output for commit: 95cb8897c790bebc40a09b9c6e9c07614f2101ab available at: https://gist.github.com/15948b6eaeffda210081<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5848.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5848.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: ec65bfc43d1135790286d6afd0dff5bb446f2aa5 available at: https://gist.github.com/15948b6eaeffda210081<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5848.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5848.dmg

**Comment by ZappoMan** [comment]
After discussing this further, we've decided to not merge this PR into our main repo, there are a couple of reasons for this decision:

* We don't want to directly merge another open source project into our code base, if we want this to be available as a regular resource, we'd rather set it up as an external dependency.
* This doesn't actually improve performance of any of our existing JS code, since none of it is using this library.
* We can get similar performance improvements in our exiting JS code by exposing our "existing APIs" as a JS implementation and baking that into the automatically included scripts for any script user.

**Pull request closed without merging**